Document/0x05j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -14,60 +14,72 @@ In the following section, we list some common root detection methods you'll enco
 
-The following is a sample attestation result:
+2.App Integrity: This section provides information about the integrity of the app, including the result of the verification (denominated "verdict"), which indicates whether the app's installation source is trusted (via Play Store) or unknown/suspicious. If the installation source is considered secure, the app version will also be displayed.
 
-```json
-{
-  "nonce": "R2Rra24fVm5xa2Mg",
-  "timestampMs": 9860437986543,
-  "apkPackageName": "com.package.name.of.requesting.app",
-  "apkCertificateDigestSha256": ["base64 encoded, SHA-256 hash of the
-                                  certificate used to sign requesting app"],
-  "apkDigestSha256": "base64 encoded, SHA-256 hash of the app's APK",
-  "ctsProfileMatch": true,
-  "basicIntegrity": true,
-}
-```
+3.Account Details: This category provides information about the app licensing status. The result can be "LICENSED", indicating that the user purchased or installed the app on the Google Play Store; "UNLICENSED", meaning that the user does not own the app or did not acquire it through the Google Play Store; or "UNEVALUATED", which means that the licensing details could not be evaluated because a necessary requirement is missing, that is, the device may not be trustworthy enough or the installed app version is not recognized by the Google Play Store.
 
-##### ctsProfileMatch Vs basicIntegrity
+4.Device Integrity: This section presents information that verifies the authenticity of the Android environment in which the app is running.
 
-The SafetyNet Attestation API initially provided a single value called `basicIntegrity` to help developers determine the integrity of a device. As the API evolved, Google introduced a new, stricter check whose results appear in a value called `ctsProfileMatch`, which allows developers to more finely evaluate the devices on which their app is running.
+- MEETS_DEVICE_INTEGRITY: Running on Android Device with Google Play Services: The app is running on an Android device with Google Play Services. This device passed system integrity checks and meets the Android compatibility requirements.
+- MEETS_BASIC_INTEGRITY: Running on Device without Google Play Services: The app is running on a device that passed basic system integrity checks. This may occur because the device may be running an unrecognized version of Android, has an unlocked bootloader, or has not been certified by the manufacturer.
+- MEETS_STRONG_INTEGRITY: The app is running on an Android device equipped with Google Play Services technology, ensuring a strong system integrity such as a hardware-protected boot process.
+- MEETS_VIRTUAL_INTEGRITY: Running on Emulator with Google Play Services: In this scenario, the app is running in a simulated environment of an Android device that has Google Play Services. The emulator was checked and approved for system integrity and meets the necessary Android compatibility requirements.

```

Comment:
Consider putting `MEETS_DEVICE_INTEGRITY`, `MEETS_BASIC_INTEGRITY`, `MEETS_STRONG_INTEGRITY` and `MEETS_VIRTUAL_INTEGRITY` in `code`

Reply:
Fixed as requested